### ISSUE TITLE
"Compare with title" option ignored

### ISSUE THREAD

**Issue by florian-schweiger** [Peuj/duplicate-tabs-closer#52]
There is an inconsistency in the duplicate detection logic between automatic and manual closing mode.

This is something I noticed with the Office365 Outlook web app where the email app lives under `https://outlook.office.com/mail/*` but the exact path depends on what folder is selected (`/mail/inbox`, `/mail/drafts`, etc.) and will also have an `/id/<SOME_ID>` appended to it in certain cases. I'll use the Outlook example to illustrate the issue.

Assume you have two tabs that both show the email inbox but differ in their path. The only practical way to identify the two as duplicates is by comparing their titles.

    | `document.URL` | `document.title`
--- | --- | ---
Existing tab | `https://outlook.office.com/mail/inbox/id/B45G3F33G8N5` | "Email - Your Name - Outlook"
Newly opened tab | `https://outlook.office.com/mail/inbox/` | "Email - Your Name - Outlook"

Now, with the settings below ("Compare with title" as sole matching rule; duplicate tabs to be closed automatically) the newly opened tab is *not* automatically closed! In the pop-up window the duplicates are listed correctly though, and clicking the "Close all" button does close the new tab as expected. So does the Alt+⇧+W command.

![image](https://user-images.githubusercontent.com/15267067/105364435-10172700-5bf5-11eb-83c5-b5fe010aca97.png)

**Comment by florian-schweiger** [comment]
Come to think about it, the logic error is actually the other way round!

With the above settings, the two tabs should **not** be identified as duplicates if the matching rules apply in an AND fashion. 
To get the result I would like, I'd have to select "Ignore path part in URL" AND "Compare with title". 

So in that sense, the rules are applied correctly in automatic closing mode, and it is the "Duplicate tabs" list that is wrong. In my example with only "Compare with title" enabled, the list should be empty – but it isn't!

On the other hand when both "Compare with title" and "Ignore path part in URL" are on, the list should not contain additional tabs that share the same host but have different titles but, and I just checked, it does! 

So in summary, it seems that the "Compare with title" rule is ignored when populating the Duplicate tabs list.

**Comment by florian-schweiger** [comment]
After some more investigating, it seems that the page title is ignored in automatic mode too. 

With `Ignore path part in URL=ON` and `Compare with title=ON`, try to open both the following links:
Link | document.title 
---|--- 
https://www.google.com/ | "Google"
https://www.google.com/maps | "Google Maps"

They are falsely detected as duplicates and one is automatically closed if that setting is active.

**Comment by Peuj** [comment]
Thanks for the investigation. I'll look at the problem.

**Comment by florian-schweiger** [comment]
I clearly misunderstood the compare title logic (see #53). A better way to address my initial problem with the Outlook web app would be to have partial path matching (see #54).
